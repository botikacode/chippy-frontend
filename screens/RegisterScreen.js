import React, {useEffect, useState} from 'react'
import { View, StyleSheet, TouchableOpacity, Text, TextInput } from 'react-native'
import Logo from '../components/Logo'
import Layout from '../components/Layout'
import SwitchSelector from "react-native-switch-selector"
import { emailValidator } from '../helpers/emailValidator'
import { passwordValidator } from '../helpers/passwordValidator'
import { nameValidator } from '../helpers/nameValidator'
import {getCustomers, saveCustomer} from '../db/customersApi'


export default function RegisterScreen({ navigation, route}) {

  const [name, setName] = useState({ value: '', error: '' })
  const [email, setEmail] = useState({ value: '', error: '' })
  const [password, setPassword] = useState({ value: '', error: '' })
  const [userType, setUserType] = useState(false)

  const userTypeSelector = [
      { label: "Personal", value: false },
      { label: "Organización", value: true }
    ];

  const handleSubmit = async (customer) => {
    try {
      await saveCustomer(customer);
      navigation.navigate("Dashboard");
    } catch (error) {
      console.log(error);
    }
  };


  const onSignUpPressed = () => {
    const nameError = nameValidator(name.value)
    const emailError = emailValidator(email.value)
    const passwordError = passwordValidator(password.value)
    if (emailError || passwordError || nameError) {
      setName({ ...name, error: nameError })
      setEmail({ ...email, error: emailError })
      setPassword({ ...password, error: passwordError })
      return
    }

    const custom = {
    lastName: "",
    firstName: name.value,
    address: "",
    city: "",
    image: "",
    description: "",
    phone: "",
    email: email.value,
    password: password.value,
    web: "",
    isShelter: userType,
  }

    handleSubmit(custom)
    navigation.navigate('Dashboard')
  }

  return (
    <Layout>
      <Logo />
      <Text style={styles.crearCuenta} >Crear Cuenta</Text>

      <View style={styles.container}>

      <SwitchSelector style={styles.switch}
        options = {userTypeSelector}
        initial={0}
        onPress={value => setUserType(value)}
        options={[
          style: {

          }
        ]}
        //onPress = {value => alert(value)}
        buttonColor ='#0094FF'
        />

      <TextInput
        style={styles.input}
        placeholder="Nombre"
        placeholderTextColor="#576574"
        label="Nombre"
        returnKeyType="next"
        value={name.value}
        onChangeText={(text) => setName({ value: text, error: '' })}
        error={!!name.error}
        errorText={name.error}
      />
      {name.error ? <Text style={styles.error}>{name.error }</Text> : null}

      <TextInput
        style={styles.input}
        placeholder="Email"
        placeholderTextColor="#576574"
        label="Email"
        returnKeyType="next"
        value={email.value}
        onChangeText={(text) => setEmail({ value: text, error: '' })}
        error={!!email.error}
        errorText={email.error}
        autoCapitalize="none"
        autoCompleteType="email"
        textContentType="emailAddress"
        keyboardType="email-address"
      />
      {email.error ? <Text style={styles.error}>{email.error }</Text> : null}

      <TextInput
        style={styles.input}
        placeholder="Contraseña"
        placeholderTextColor="#576574"
        label="Contraseña"
        returnKeyType="done"
        value={password.value}
        onChangeText={(text) => setPassword({ value: text, error: '' })}
        error={!!password.error}
        errorText={password.error}
        secureTextEntry
      />
      {password.error ? <Text style={styles.error}>{password.error }</Text> : null}

       </View>
      <TouchableOpacity style={styles.buttonCeleste}  onPress={onSignUpPressed}>
          <Text style={styles.buttonText}>Aceptar</Text>
      </TouchableOpacity>

    </Layout>
  )
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    marginTop: 4,
  },
  link: {
    fontWeight: 'bold',
  },
  crearCuenta:{
    color: "#0094FF",
    marginBottom: 10,
    fontSize: 20,
  },
  input: {
    width: "70%",
    marginBottom: 4,
    marginTop: 10,
    fontSize: 14,
    borderWidth: 1,
    borderColor: "#ced4da",
    height: 30,
    color: "#000000",
    padding: 4,
    borderRadius: 5,
  },
  buttonCeleste: {
    paddingTop: 10,
    marginTop: 24,
    paddingBottom: 10,
    borderRadius: 5,
    marginBottom: 3,
    backgroundColor: "#0094FF",
    width: "50%",
  },
  buttonText: {
    color: "#fff",
    textAlign: "center",
  },
  container: {
    width: '100%',
    marginVertical: 5,
    alignItems: 'center',
  },
  error: {
    fontSize: 10,
    color: '#f13a59',
    paddingBottom:5,
  },

})
