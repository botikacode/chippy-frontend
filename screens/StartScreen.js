import React from 'react'
import Logo from '../components/Logo'
import Layout from '../components/Layout'

import { TextInput, TouchableOpacity, Text, StyleSheet } from "react-native";
export default function StartScreen({ navigation }) {
  return (
    <Layout>
      <Logo />
      <Text style={styles.welcome} >Â¡Bienvenido a Chippy!</Text>
      <TouchableOpacity style={styles.buttonCeleste} onPress={() => navigation.navigate("LoginScreen")}>
          <Text style={styles.buttonText}>Ya tengo una cuenta</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.buttonBlanco} onPress={() => navigation.navigate("RegisterScreen")}>
          <Text style={styles.buttonTextCeleste}>Registrarme</Text>
      </TouchableOpacity>

      {/* 
      <Button
        mode="contained"
        onPress={() => navigation.navigate('LoginScreen')}
      >
        Ya tengo una cuenta
      </Button>
      <Button
        mode="outlined"
        onPress={() => navigation.navigate('RegisterScreen')}
      >
        Registrarme
      </Button> */}

    </Layout>
  )
}

const styles = StyleSheet.create({
  
  buttonCeleste: {
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 5,
    marginBottom: 3,
    backgroundColor: "#0094FF",
    width: "90%",
  },
  buttonBlanco: {
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 5,
    marginBottom: 3,
    backgroundColor: "#fff",
    width: "90%",
  },
  buttonText: {
    color: "#fff",
    textAlign: "center",
  },
  buttonTextCeleste: {
    color: "#0094FF",
    textAlign: "center",
  },
  welcome:{
    color: "#0094FF",
    marginBottom: 20,
    fontSize: 16,
    fontWeight: 'bold',
  },
  
})