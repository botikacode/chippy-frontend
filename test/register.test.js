import React from "react"
import renderer from "react-test-renderer"
import { render, screen, fireEvent } from "@testing-library/react-native"
// import '@testing-library/jest-dom'
import RegisterScreen from "../screens/RegisterScreen"
import StartScreen from "../screens/StartScreen"
import { Alert } from "react-native"

test("Renderiza Register Screen correctamente", () => {
  const tree = renderer.create(<RegisterScreen/>).toJSON()
  expect(tree).toMatchSnapshot()
});

test("Se registra bien como usuario", async () => {
  jest.spyOn(Alert, "alert")

  render(<RegisterScreen/>)

  const name = screen.getByPlaceholderText("Nombre")
  const email = screen.getByPlaceholderText("Email")
  const password = screen.getByPlaceholderText("Contrase√±a")

  fireEvent.changeText(name, "Pepe")
  fireEvent.changeText(email, "pepeusuario@gmail.com")
  fireEvent.changeText(password, "12345678")

  expect(name.props.value).toBe("Pepe")
  expect(email.props.value).toBe("pepeusuario@gmail.com")
  expect(password.props.value).toBe("12345678")

  // Hace click en Aceptar
  fireEvent.press(screen.getByText("Aceptar"))

  expect(Alert.alert)
});
  