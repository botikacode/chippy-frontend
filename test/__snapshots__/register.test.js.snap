// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderiza Register Screen correctamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/logo.png",
      }
    }
    style={
      {
        "height": 110,
        "marginBottom": 10,
        "minWidth": 200,
        "width": 110,
      }
    }
  />
  <Text
    style={
      {
        "color": "#0094FF",
        "fontSize": 20,
        "marginBottom": 10,
      }
    }
  >
    Crear Cuenta
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "marginVertical": 5,
        "width": "100%",
      }
    }
  >
    <TextInput
      error={false}
      errorText=""
      label="Nombre"
      onChangeText={[Function]}
      placeholder="Nombre"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 4,
          "marginTop": 10,
          "padding": 4,
          "width": "70%",
        }
      }
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCompleteType="email"
      error={false}
      errorText=""
      keyboardType="email-address"
      label="Email"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#576574"
      returnKeyType="next"
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 4,
          "marginTop": 10,
          "padding": 4,
          "width": "70%",
        }
      }
      textContentType="emailAddress"
      value=""
    />
    <TextInput
      error={false}
      errorText=""
      label="Contraseña"
      onChangeText={[Function]}
      placeholder="Contraseña"
      placeholderTextColor="#576574"
      returnKeyType="done"
      secureTextEntry={true}
      style={
        {
          "borderColor": "#ced4da",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "#000000",
          "fontSize": 14,
          "height": 30,
          "marginBottom": 4,
          "marginTop": 10,
          "padding": 4,
          "width": "70%",
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#0094FF",
        "borderRadius": 5,
        "marginBottom": 3,
        "marginTop": 24,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingTop": 10,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      Aceptar
    </Text>
  </View>
</View>
`;
